Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericTestParameterConverter.cs

```diff
@@ -1,69 +1,105 @@
-﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// This file is part of the re-motion Core Framework (www.re-motion.org)
 // Copyright (c) rubicon IT GmbH, www.rubicon.eu
-// 
-// The re-motion Core Framework is free software; you can redistribute it 
-// and/or modify it under the terms of the GNU Lesser General Public License 
-// as published by the Free Software Foundation; either version 2.1 of the 
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
 // License, or (at your option) any later version.
-// 
-// re-motion is distributed in the hope that it will be useful, 
-// but WITHOUT ANY WARRANTY; without even the implied warranty of 
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 // GNU Lesser General Public License for more details.
-// 
+//
 // You should have received a copy of the GNU Lesser General Public License
 // along with re-motion; if not, see http://www.gnu.org/licenses.
-// 
+//
 using System;
-using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using System.Web.Script.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 // ReSharper disable once CheckNamespace
 
 namespace Remotion.Web.Development.WebTesting.IntegrationTests.Infrastructure
 {
-  public class GenericTestParameterConverter : JavaScriptConverter
+  public class GenericTestParameterConverter : JsonConverter<GenericTestPageParameterDto>
   {
-    public static readonly GenericTestParameterConverter Instance = new GenericTestParameterConverter();
-
-    /// <inheritdoc />
-    public override IEnumerable<Type> SupportedTypes { get; } = new[] { typeof (GenericTestPageParameterDto) };
+    public static readonly GenericTestParameterConverter Instance = new();
 
     private GenericTestParameterConverter ()
     {
     }
 
-    /// <inheritdoc />
-    public override object Deserialize (IDictionary<string, object> dictionary, Type type, JavaScriptSerializer serializer)
+    public override GenericTestPageParameterDto Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
I didn't actually play with the code, but couldn't we have done something like 
`dto = JsonSerializer.Deserialize<Dictionary<string, object>> (ref reader, options);`
and then
`var status = (int)dto[status]`
`var parameters = (IDictionary<string, IList<string>)dto[parameters]`

to have less of gritty details? The code appears to be correct and very safe, it just feels like there's a lot of work for 'just test infrastructure' here.

Reply:
@MichaelKetting  Doing this would result in a `Dictionary<string, JsonElement>` typed as `Dictionary<string, object>`, the dictionary values would still need some kind of extraction/manipulation.

Reply:
@iam-mholle okay, so we make it `Dictionary<string, JsonElement>`? Still appears to be the more compact, less detail-y way of doing this?

Reply:
Refer to [`cc9bd8e` (#114)](https://github.com/re-motion/Framework/pull/114/commits/cc9bd8e68ef58232f36161b59e024fe98aa0ec66) for the updated implementation.

Reply:
I think it looks better. In theory, we could keep the original code structure with the helper methods, but I don't mind either way.

For serialize: did you try doing the dictionary-to-json approach from the original code base?

Reply:
That would work too, I guess its easier to understand and maintain, I just liked the other approach a bit more but I don't mind changing it.